src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,114 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.DueCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+
+public class Parser {
+    private static final String TODO = "todo";
+    private static final String DELETE = "delete";
+    private static final String DONE = "done";
+    private static final String EVENT = "event";
+    private static final String DEADLINE = "deadline";
+    private static final String LIST = "list";
+    private static final String DUE = "due";
+    private static final String FIND = "find";
+
+    public static ArrayList<String> parseToStart(ArrayList<String> oldData) {
+        ArrayList<String> parsedData = new ArrayList<>();
+        for (String s : oldData) {
+            String[] array = s.split(" \\| ");
+            if (s.contains("T")) {
+                array[0] = "todo";
+            } else if (s.contains("D")) {
+                array[0] = "deadline";
+            } else {
+                array[0] = "event";
+            }
+            String parsedString = array[0];
+            for (int i = 1; i < array.length; i++) {
+                if (i == 3) {
+                    parsedString = array[0].equals("deadline") ? parsedString + " /by " + array[i] : parsedString
+                            + " /at " + array[i];
+                } else {
+                    parsedString = parsedString + " " + array[i];
+                }
+            }
+            parsedData.add(parsedString);
+        }
+        return parsedData;
+    }
+
+    public static Command parse(String input) throws DukeException {
+        if (input.equals("")) {
+            throw new DukeException("No command typed! Please key in a valid command.");
+        } else {
+            String[] inputArray = input.split(" ", 2);
+            String command = inputArray[0];
+            if (command.equals(DELETE) || command.equals(DONE) || command.equals(DUE) || command.equals(FIND)) {
+                if (inputArray.length == 1) {
+                    throw new DukeException("Missing argument! Please try again.");
+                } else if (inputArray.length > 2) {
+                    throw new DukeException("Additional arguments! Please try again.");
+                } else {
+                    try {
+                        //Check if the second argument is a valid integer
+                        if (command.equals(DONE) || command.equals(DELETE)) {
+                            Integer.parseInt(inputArray[1]);
+                            return command.equals(DELETE) ? new DeleteCommand(inputArray) : new DoneCommand(inputArray);
+                        } else if (command.equals(DUE)) {
```

Comment:
Perhaps all the nested-if else statements are a bit difficult to read and follow? Maybe you can try using switch statements?